### PR TITLE
Docs: Document fine-grained access control

### PR DESCRIPTION
**Pull request by vtorosyan**
**What this PR does / why we need it**:

The PR introduces new docs for the fine-grained access control and updates relevant connected pages. The [issue](https://github.com/grafana/grafana-enterprise/issues/1282) describes the scope of this PR. [This comment](https://github.com/grafana/grafana/pull/33563#issuecomment-833501418) contains few screenshots.

**Note:** If you know any page that needs to be updated or a new documentation to be added, please update the issue so that we can keep track of changes.

**Big Question**

At the moment, I'm using "Access Control" name for the new feature. However, I consider Fine-grained access control as a better alternative. Main reasoning behind it is that "access control" is a clearly overused wording and it is very hard to distinguish which access control we mean when talking about it in the new docs. While I see us eventually moving towards having one unified Access Control, thus naming the capability like that, now we are far away from that.

We have also covered this a bit in the [Slack conversation](https://raintank-corp.slack.com/archives/C01LU2Y5527/p1620295553224800).

**UPDATE:** I have decided to switch the term to `Fine-grained access control`, for the above mentioned reasons. Please do let me know what you think about it and if there are any concerns or suggestions. Thanks~

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/1282

### CODE REVIEW

File: docs/sources/enterprise/access-control/concepts/roles.md
Comment (gamab) [inline docs/sources/enterprise/access-control/concepts/roles.md:48]:
```suggestion
To control what your users can access or not, you can assign or revoke a role to the existing Organization roles, or to a Grafana Admin. These assignments are called built-in role assignments.
```

File: docs/sources/http_api/access_control.md
Comment (gamab) [inline docs/sources/http_api/access_control.md:532]:
```suggestion
200 | Role was revoked from built-in role.
```

**Comment by vtorosyan** [comment]
Adding couple of screenshots to give an idea about how changes look like.

**Nav menu**
Screenshot 2021-05-06 at 14 55 38

**Roles page with all predefined roles**
Screenshot 2021-05-06 at 14 55 52

**Permissions**
Screenshot 2021-05-06 at 14 56 07

**HTTP API parts**
Screenshot 2021-05-06 at 14 56 27

**Banner with note**
Screenshot 2021-05-06 at 14 56 33

**Comment by vtorosyan** [comment]
I am opening this up for review. While there is a lot to improve, I believe it's in a good place to start getting feedback.

**Review by sakjur — COMMENTED** [review]
🎉 This is great work!

I left a bunch of comments where the most important ones are places where I've found the documentation to be slightly inaccurate/incomplete.

**Review by oddlittlebird — DISMISSED** [review]
Wow, lots of great content in this PR! When @osg-grafana is back tomorrow, let's discuss how to make it even better.

File: docs/sources/enterprise/access-control/_index.md
Comment (mjseaman) [reply docs/sources/enterprise/access-control/_index.md:14]: Agree with @vtorosyan's sentiment. Perhaps where we use the term fine-grained access control other than the title and definition, we can find more specific terms like "custom roles" or "updating permissions for a role."

File: docs/sources/enterprise/access-control/_index.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/_index.md:8]: This is a bit tricky to do, as essentially everyone who has required permissions can manage access control. That said, when Grafana starts, in the beginning only Grafana Server Admin can do that, and Grafana Server Admin can later decide that someone else needs also to be able to do it and give them permissions.

I will look into where it would make sense to mention this somehow.

File: docs/sources/enterprise/access-control/provisioning.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/provisioning.md:98]: Not sure, this was from Gabriel from the provisioning work. I will try to test.

File: docs/sources/enterprise/access-control/concepts/permissions.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/concepts/permissions.md:8]: It is not applicable at all to OSS Grafana from usage and product perspective. That said, a lot of code is in the OSS Grafana, so it might be valuable to cover some parts of this in a developer guide. That would be out of scope of this PR though.

File: docs/sources/enterprise/access-control/concepts/permissions.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/concepts/permissions.md:14]: They don't have to be, I'll remove.

File: docs/sources/enterprise/access-control/concepts/permissions.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/concepts/permissions.md:25]: Just to make sure that I understand right, you mean the heading should be H1 not H2? 🤔

File: docs/sources/enterprise/access-control/concepts/roles.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/concepts/roles.md:18]: I did not want to put in Scope, as we have a scope concept for permission, and this one is different. I understand the confusion, I will think how to improve this.

File: docs/sources/enterprise/access-control/_index.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/_index.md:17]: Honestly the easiest "fix" would be to remove this hierarchy and have flat structure, so roles, permissions and custom roles will be in the same level as the index 🤔

File: docs/sources/enterprise/access-control/how-to-examples.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/how-to-examples.md:8]: Yes it's a todo to add more examples.

**Comment by vtorosyan** [comment]
Thanks everyone for contributions and suggestions. Great feedback!
I've tried to touch all raised concerns and unfortunately this means a major refactoring. I'd recommend to review from start, rather then new commits only.

Major changes:
- Structure is flattened. I have removed "Concepts" subsection, "Managing roles and permissions" section and simplified things a lot.
- Tried to follow https://github.com/grafana/grafana/tree/main/contribute/templates#documentation-templates whenever possible

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:3]:
```suggestion
description = "Grant, change, or revoke access to Grafana resources"
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:10]:
```suggestion
> **Note:** Fine-grained access control is in beta, and you can expect changes in future releases.
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:12]:
```suggestion
Fine-grained access control provides a standardized way of granting, changing, and revoking access when it comes to viewing and modifying Grafana resources, such as users and reports. 
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:13]:
```suggestion
Fine-grained access control works with [Grafana permissions]({{< relref "../../permissions/_index.md" >}}), and it allows you granular control of users’ actions.
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:16]:
```suggestion
To use the fine-grained access control system, refer to [Fine-grained access control usage scenarios]({{< relref "./usage-scenarios.md" >}}).
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:21]:
```suggestion
To enable it in different environments, refer to [Configuration]({{< relref "../../administration/configuration.md" >}}).
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:27]:
```suggestion
Fine-grained access control considers a) _who_ has an access (`identity`), and b) _what they can do_ and on which _Grafana resource_ (`role`).
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:29]:
```suggestion
You can grant, change, or revoke access to _users_ (`identity`). When an authenticated user tries to access a Grafana resource, the authorization system checks the required fine-grained permissions for the resource and determines whether or not the action is allowed. Refer to [Fine-grained permissions]({{< relref "./permissions.md" >}}).
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:31]:
```suggestion
To grant or revoke access to your users, create or remove built-in role assignments. For more information, refer to [Built-in role assignments]({{< relref "./roles.md#built-in-role-assignments" >}}).
```

Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:31]:
```suggestion
To grant or revoke access to your users, create or remove built-in role assignments. For more information, refer to [Built-in role assignments]({{< relref "./roles.md#built-in-role-assignments" >}}).
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:35]:
```suggestion
Fine-grained access control permissions are available for a subset of resources. For more information, refer to [Permissions]({{< relref "./permissions.md" >}}).
```

File: docs/sources/enterprise/access-control/_index.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/_index.md:42]:
```suggestion
To learn more about specific endpoints where you can use access control, refer to the relevant API guide:

- [Access Control API]({{< relref "../../http_api/access_control.md" >}})
- [Admin API]({{< relref "../../http_api/admin.md" >}})
- [Organization API]({{< relref "../../http_api/org.md" >}})
- [Reporting API]({{< relref "../../http_api/reporting.md" >}})
- [User API]({{< relref "../../http_api/user.md" >}})
```

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:3]:
```suggestion
description = "Understand fine-grained access control permissions"
```
As odd as it is, the weight (importance) falls on the final word in this phrase.

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:11]:
```suggestion
Each permission is defined by an action and a scope. When evaluating a fine-grain access decision, consider what specific action a user will be allowed to perform and on what resource.

To grant permissions to a user, create built-in role assignments. For more information, refer to [Built-in role assignments]({{< relref "./roles.md#built-in-role-assignments" >}}).
```

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:13]:
```suggestion
To learn more about what permissions are used for what resources, refer to [Resources with fine-grained permissions]({{< relref "./_index.md#resources-with-fine-grained-permissions" >}}).
```
Write "what" when considering more than two options in the context of a choice. Write "which" when the choice involves two things.

**Review by osg-grafana — CHANGES_REQUESTED** [review]

File: docs/sources/enterprise/access-control/_index.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/_index.md:10]: One day, would be nice to update these in other docs. Since I suspect many people, including me are just looking at the previous examples and copy pasting it.

File: docs/sources/enterprise/access-control/permissions.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/permissions.md:13]: 🚀

**Comment by vtorosyan** [comment]
I believe I am done adding with relevant content. From now on, hopefully there will only be edits.

**Comment by vtorosyan** [comment]
A friendly reminder that we are a bit tight with timeline with this PR. Beta2 goes out on Thursday, 20th of May, ideally we will have docs pushed, if not, still not the end of the world I guess, we would need to get this though before June.

**Review by osg-grafana — CHANGES_REQUESTED** [review]

**Comment by osg-grafana** [comment]
> A friendly reminder that we are a bit tight with timeline with this PR. Beta2 goes out on Thursday, 20th of May, ideally we will have docs pushed, if not, still not the end of the world I guess, we would need to get this though before June.

Yes, and involve your friendly neighborhood technical writer early and often. Looking forward to pairing tomorrow morning.

**Comment by vtorosyan** [comment]
As discussed in a group, we will be merging this PR as it is and continue working to improve the docs incrementally.

**Review by osg-grafana — APPROVED** [review]
Will be updating docs next week. Approving to unblock this for beta2.

**Review by osg-grafana — APPROVED** [review]
Here you go, @vtorosyan. Have a nice long weekend. You deserve it.

**Comment by vtorosyan** [comment]
@oddlittlebird I'd appreciate if you can re-accept the PR, otherwise it is blocked for the merge. We will be working on the docs update next week, as we decided to ship the first version of docs for beta. Thanks!

**Pull request merged**
